user_tools/src/spark_rapids_pytools/rapids/rapids_job.py

```diff
@@ -125,11 +129,55 @@ def run_job(self):
 @dataclass
 class RapidsLocalJob(RapidsJob):
     """
-    Implementation of a RAPIDS job that runs local on a local machine.
+    Implementation of a RAPIDS job that runs local on a machine.
     """
 
-    def _build_classpath(self):
+    def _get_hadoop_classpath(self) -> Optional[str]:
+        """
+        Gets the Hadoop's configuration directory from the environment variables.
+        The first valid directory found is returned in the following order:
+        1. HADOOP_CONF_DIR
+        2. HADOOP_HOME/conf
+        3. HADOOP_HOME/etc/hadoop
+        Otherwise, returns None.
+
+        """
+        hadoop_dir_lookups = {
+            'hadoopConfDir': {
+                'envVar': 'HADOOP_CONF_DIR',
+                'postfix': ''
+            },
+            'hadoopHomeV1': {
+                'envVar': 'HADOOP_HOME',
+                'postfix': '/conf'
+            },
+            'hadoopHomeV2': {
+                'envVar': 'HADOOP_HOME',
+                'postfix': '/etc/hadoop'
+            }
+        }
+        # Iterate on the hadoop_dir_lookups to return the first valid directory found.
+        for dir_key, dir_value in hadoop_dir_lookups.items():
+            env_var_value = Utils.get_sys_env_var(dir_value['envVar'])
+            if env_var_value is not None:
+                postfix = dir_value['postfix']
+                conf_dir = f'{env_var_value}{postfix}'
+                try:
+                    conf_dir_path = LocalPath(conf_dir)
+                    if conf_dir_path.is_dir() and conf_dir_path.exists():
+                        # return the first valid directory found without the URI prefix
+                        return conf_dir_path.no_prefix
+                except Exception as e:  # pylint: disable=broad-except
+                    self.logger.debug(
+                        'Could not build hadoop classpath from %s. Reason: %s', dir_key, e)
+        return None
+
+    def _build_classpath(self) -> List[str]:
         deps_arr = [self.prop_container.get_jar_file()]
+        hadoop_cp = self._get_hadoop_classpath()
+        # append hadoop conf dir if any
+        if hadoop_cp is not None:
```

Comment:
Should this throw an exception when hadoop conf dir was not found but event logs are in hdfs?